src/components/ChooseCurrency.tsx

```diff
@@ -1,119 +1,36 @@
+    Currency,
+    eify,
+    FIAT_OPTIONS,
+    timeout
+} from "~/utils";
 
 type ChooseCurrencyForm = {
     fiatCurrency: string;
 };
 
-/**
- *  FIAT_OPTIONS is an array of possible currencies
- *  All available currencies can be found here https://api.coingecko.com/api/v3/simple/supported_vs_currencies
- *  @Currency
- *  @param {string} label - should be in the format {Name} {ISO code}
- *  @param {string} values - are uppercase ISO 4217 currency code
- *  @param {string?} hasSymbol - if the currency has a symbol it should be represented as a string
- *  @param {number} maxFractionalDigits - the standard fractional units used by the currency should be set with maxFractionalDigits
- *
- *  Bitcoin is represented as:
- *  {
- *      label: "bitcoin BTC",
- *      value: "BTC",
- *      hasSymbol: "₿",
- *      maxFractionalDigits: 8
- *  }
- */
-
-export const FIAT_OPTIONS: Currency[] = [
-    {
-        label: "Bitcoin BTC",
-        value: "BTC",
-        hasSymbol: "₿",
-        maxFractionalDigits: 8
-    },
-    {
-        label: "United States Dollar USD",
-        value: "USD",
-        hasSymbol: "$",
-        maxFractionalDigits: 2
-    },
-    { label: "Swiss Franc CHF", value: "CHF", maxFractionalDigits: 2 },
-    {
-        label: "Chinese Yuan CNY",
-        value: "CNY",
-        hasSymbol: "¥",
-        maxFractionalDigits: 2
-    },
-    {
-        label: "Euro EUR",
-        value: "EUR",
-        hasSymbol: "€",
-        maxFractionalDigits: 2
-    },
-    {
-        label: "Brazilian Real BRL",
-        value: "BRL",
-        hasSymbol: "R$",
-        maxFractionalDigits: 2
-    },
-    {
-        label: "British Pound GBP",
-        value: "GBP",
-        hasSymbol: "₤",
-        maxFractionalDigits: 2
-    },
-    {
-        label: "Australia Dollar AUD",
-        value: "AUD",
-        hasSymbol: "$",
-        maxFractionalDigits: 2
-    },
-    {
-        label: "Japanese Yen JPY",
-        value: "JPY",
-        hasSymbol: "¥",
-        maxFractionalDigits: 0
-    },
-    {
-        label: "Korean Won KRW",
-        value: "KRW",
-        hasSymbol: "₩",
-        maxFractionalDigits: 0
-    },
-    { label: "Kuwaiti Dinar KWD", value: "KWD", maxFractionalDigits: 3 }
-].sort((a, b) => (a.value > b.value ? 1 : b.value > a.value ? -1 : 0));
-
-export const USD_INDEX = FIAT_OPTIONS.findIndex((fo) => fo.value === "USD");
-export const BTC_INDEX = FIAT_OPTIONS.findIndex((fo) => fo.value === "BTC");
+const COMBINED_OPTIONS: Currency[] = BTC_USD_OPTIONS.concat(FIAT_OPTIONS);
 
 export function ChooseCurrency() {
     const i18n = useI18n();
     const [error, setError] = createSignal<Error>();
-    const [state, actions] = useMegaStore();
+    const [_state, actions] = useMegaStore();
     const [loading, setLoading] = createSignal(false);
     const navigate = useNavigate();
 
     function findCurrencyByValue(value: string) {
-        return FIAT_OPTIONS.find((currency) => currency.value === value);
+        return (
```

Comment:
reverted this previous change from #575 realized that behavior was not ideal

Reply:
Tried to get the button disabled when `!chooseCurrencyForm.touched` but it wasn't working properly

Reply:
no worries disabled buttons states are overrated